Assets/_BForBoss/_Entities/Scripts/LifeCycleBehaviour.cs

```diff
@@ -17,6 +18,11 @@ public abstract class LifeCycleBehaviour : MonoBehaviour
 
         public bool IsAlive => _lifeCycle.IsAlive;
 
+        public void NotifyOnDeath(LifeCycleManager manager)
+        {
+            _lifeCycle.NotifyOnDeath(manager);
```

Comment:
Not really a review ccomment but just random thoughts. 
The reason why each individual LifeCycle bhaviour/C# object needs to know about the manager. Is because the life cycle behaviour creates their own `new LifeCycle(_health);`

If the manager Initializes each `LifeCycleBehaviour` and DI a `new LifeCycle(_health);` into each one, then you'll be able to subscribe the manager onto each individual entity, instead of having the entity be depenent on a manager.

Reply:
Hmm, I can see that. Definitely would like to bring this up on Tuesday too though for discussion